src/htmx.js

```diff
@@ -1060,6 +1061,15 @@ return (function () {
                 triggerErrorEvent(elt, "htmx:wsError", {error:e, socket:socket});
                 maybeCloseWebSocketSource(elt);
             };
+
+            socket.onclose = function (e) {
+                if ([1006, 1012, 1013].includes(e.code)) {  // Abnormal Closure/Service Restart/Try Again Later
+                    setTimeout(function() {
+                        processWebSocketSource(elt, wssSource);  // creates a websocket with a new timeout
```

Comment:
will this retry on a failure to reconnect?

Reply:
Yes. In this case it will be a abnormal closure (1006). Verified in Chrome and Firefox.